cogboard-webapp/src/hooks/index.js

```diff
@@ -28,7 +35,10 @@ export const useFormData = (data) => {
       return;
     }
 
-    setValues({ ...values, [propName]: fieldValue});
+    const newValues = { ...values, [propName]: fieldValue};
```

Comment:
What about the case for nested object values? There is return above without any validation.